tools/createNewRollup/createNewRollup.ts

```diff
@@ -0,0 +1,399 @@
+    ) {
+        // Get token metadata
+        gasTokenMetadata = await rollupBridgeContract.getTokenMetadata(createRollupParameters.gasTokenAddress);
+        outputJson.gasTokenMetadata = gasTokenMetadata;
+        // If gas token metadata includes `0x124e4f545f56414c49445f454e434f44494e47 (NOT_VALID_ENCODING)` means there is no erc20 token deployed at the selected gas token network
+        if (gasTokenMetadata.includes("124e4f545f56414c49445f454e434f44494e47")) {
+            throw new Error(
+                `Invalid gas token address, no ERC20 token deployed at the selected gas token network ${createRollupParameters.gasTokenAddress}`
+            );
+        }
+        const wrappedData = await rollupBridgeContract.wrappedTokenToTokenInfo(createRollupParameters.gasTokenAddress);
+        if (wrappedData.originNetwork != 0n) {
+            // Wrapped token
+            gasTokenAddress = wrappedData.originTokenAddress;
+            gasTokenNetwork = wrappedData.originNetwork;
+        } else {
+            // Mainnet token
+            gasTokenAddress = createRollupParameters.gasTokenAddress;
+            gasTokenNetwork = 0n;
+        }
+    } else {
+        gasTokenAddress = ethers.ZeroAddress;
+        gasTokenNetwork = 0;
+        gasTokenMetadata = "0x";
+    }
+
+    let batchData = "";
+    // If is vanilla client, replace genesis by sovereign contracts, else, inject initialization batch
+    if (isVanillaClient) {
+        const pathGenesis = path.join(__dirname, "./genesis.json");
+        let genesis = require(pathGenesis);
+        const initializeParams = {
+            rollupID: rollupID,
+            gasTokenAddress,
+            gasTokenNetwork,
+            polygonRollupManager: ethers.ZeroAddress,
+            gasTokenMetadata,
+            bridgeManager: sovereignParams.bridgeManager,
+            sovereignWETHAddress: sovereignParams.sovereignWETHAddress,
+            sovereignWETHAddressIsNotMintable: sovereignParams.sovereignWETHAddressIsNotMintable,
+            globalExitRootUpdater: sovereignParams.globalExitRootUpdater,
+            globalExitRootRemover: sovereignParams.globalExitRootRemover,
+        };
+        genesis = await updateVanillaGenesis(genesis, chainID, initializeParams);
+        // Add weth address to deployment output if gas token address is provided and sovereignWETHAddress is not provided
+        if (
+            gasTokenAddress !== ethers.ZeroAddress &&
+            ethers.isAddress(gasTokenAddress) &&
+            (sovereignParams.sovereignWETHAddress === ethers.ZeroAddress ||
+                !ethers.isAddress(sovereignParams.sovereignWETHAddress))
+        ) {
+            const wethObject = genesis.genesis.find(function (obj: {contractName: string}) {
+                return obj.contractName == "WETH";
+            });
+            outputJson.WETHAddress = wethObject.address;
+        }
+        fs.writeFileSync(pathGenesis, JSON.stringify(genesis, null, 1));
```

Comment:
Do not overwrite the input-genesisi. create a new one.
Writing files should happen all together at the end of the script, just to not create output files if anything goes wrong with the script

Reply:
I have been thinking and I will add the genesis-sovereign to the output script file